test/cbmc/proofs/setControlInterface/setControlInterface_harness.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * AWS IoT Over-the-air Update v3.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file setControlInterface_harness.c
+ * @brief Implements the proof harness for setControlInterface function.
+ */
+/* Ota interface includes. */
+#include "ota_interface_private.h"
+
+void setControlInterface_harness()
+{
+    OtaControlInterface_t * pControlInterface;
+    OtaControlInterface_t controlInterface;
+
+    /* Initialize the control interface. The control interface can never
+     *  be NULL as it is declared statically in ota.c source file. */
+    pControlInterface = &controlInterface;
+
+    setControlInterface( pControlInterface );
+}
```

Comment:
This test seems like it would only be valiud if the configENABLED_CONTROL_PROTOCOL macro is set to OTA_CONTROL_OVER_MQTT.
https://github.com/aws/ota-for-aws-iot-embedded-sdk/blob/813dcbc93e7b1088f890ce1ca023816127f91938/source/ota_interface.c#L67-L71


Can you make it so that the test defines this macro value so we aren't just assuming that the default config file is what it needs to be?

Reply:
This won't work. You defined the macro in the scope of this .c file, so it will not be visible in the ota_interface.c file, where it needs to be used.

You need to define the macro either in a config file file, or a preprocessor definition. I don't recommend making test specific changes to a config file unless you have a way to have separate configs for each test. So you should probably use preprocessor definitions via compiler flags.

Reply:
Updated it in 2e321cf